packages/anchor-listener/src/listener.ts

```diff
@@ -0,0 +1,97 @@
+import { ANCHOR_CONTRACT_ADDRESSES } from '@ceramicnetwork/anchor-contract'
+import type { Block, Provider } from '@ethersproject/providers'
+import type { CID } from 'multiformats/cid'
+import {
+  type Observable,
+  type RetryConfig,
+  firstValueFrom,
+  fromEvent,
+  map,
+  mergeMap,
+  scan,
+} from 'rxjs'
+
+import { createRootCIDsLoader, mapLoadBlock } from './loader.js'
+
+export type ProcessedBlockEvent = {
```

Comment:
To re create the ipfs anchor proof I will need the following information: 
{
      blockNumber: tx.blockNumber,
      blockTimestamp: tx.blockTimestamp,
      root: merkleRootCid,
      chainId: tx.chain,
      txHash: txHashCid,
    }

This is going to change because of this project found here: https://linear.app/3boxlabs/project/graceful-block-reorgs-f1707041a92d. I believe we will be only relying on transaction hash. 

I possible could we provide additional information?

Reply:
Yes looks like we can get the transaction hash from the log entry (https://docs.ethers.io/v5/api/providers/types/#providers-Log) so shouldn't be an issue for this. We can get the block number as well but not the block timestamp though it seems.

Also looking at the `Log` interface from ethers, there's a `removed` flag, should we take it into account please? For example, ignoring logs that are removed?

Reply:
I think just making a call to `eth_getBlockByNumber` can be used to fetch the timestamp. 
This is a bit extra overhead, but will be removed when the [Graceful block reorgs](https://linear.app/3boxlabs/project/graceful-block-reorgs-f1707041a92d) is implemented. 

cc @stbrody

Reply:
Ah, thanks @stephhuynh18 you already pointed this out!

Reply:
> Also looking at the `Log` interface from ethers, there's a `removed` flag, should we take it into account please? For example, ignoring logs that are removed?

I am not familiar with the flag. Based on this comment: 
```
log.removed is only works with filters API e.g. [eth_getFilterChanges](https://eth.wiki/json-rpc/API#eth_getfilterchanges)

It is always false for eth_getLogs
```
https://github.com/ethers-io/ethers.js/issues/385#issuecomment-1076735812 
I don't think we can use this

Reply:
Should be addressed now with providing `AnchorProof` objects in the event.